back-end/controller/contact.js

```diff
@@ -0,0 +1,123 @@
+        });
+
+    } catch (error) {
+        next({"status": error.status, "messege": error.messege});
+    }
+}
+
+//Get All names
+exports.isNameThere = async (req, res, next) => {
+    try {
+        const { name } = req.params;
+        await Contact.find({name})
+        .then((nameData) => {
+            if (nameData.length > 0) res.status(200).send(true);
+            else res.send(false);
+        })
+        .catch((error) => {
+            next({"status": error.status, "messege": error.messege})
+        });
+
+    } catch (error) {
+        next({"status": error.status, "messege": error.messege});
+    }
+}
+
+//Get contact by id
+exports.getContactById = async (req, res, next) => {
+    try {
+        const { id } = req.params;
+        await Contact.find({_id: id})
+        .then((phoneBookContact) => res.json(phoneBookContact))
+        .catch((error) => { 
+            next({"status": 404, "messege": `bad request - no person with ID ${id}`}) 
+        });
+    } catch (error) {
+        next({"status": error.status, "messege": error.messege});
+    }
+}
+
+//Get info about phoneBook
+exports.getInfo = async (req, res, next) => {
+    try {
+        await Contact.find({})
+        .then((phoneBookData) => {
+            res.json({"peopleNum": `PhoneBook has info for ${phoneBookData.length} people`,
+                       "date": `${moment().format('llll')}`});
+        })
+        .catch( (error) => {
+            next({"status": error.status, "messege": error.messege})
+        });
+    } catch (error) {
+        next({"status": error.status, "messege": error.messege});    }
+}
+
+// Add a new contact by "name" and "number"
+exports.addContact = async (req, res, next) => {
+    try {
+        const {name, phoneNumber} = req.body;
+        //Check if one of the details has not been entered 
+        if (!name || !phoneNumber) {
+            next({"status": 400, "messege": "Must enter name and number"});
+        }
+        const contact = new Contact ({ name , number: phoneNumber});
+        contact.save()
+        .then( (savedContact) => res.status(200).send(true) )
+        .catch( (error) => {
+            next({"status": 400, "messege": "Can't add new contact, try to choose another name or number"})
+        });
+
+    } catch (error) {
+        next({"status": error.status, "messege": error.messege});
+    }
+}
+
+//Update contact phone number by name
+exports.updateContact = async (req, res, next) => {
+    try {
+        const {name, phoneNumber} = req.body;
+        //Check if one of the details has not been entered 
+        if (!name || !phoneNumber) {
+            next({"status": 400, "messege": "Must enter name and number"});
+        }
+        Contact.findOneAndUpdate({ name }, {"number": phoneNumber})
+        .then( (savedContact) => res.status(200).send(true) )
+        .catch( (error) => {
+            next({"status": 400, "messege": "Can't update contact, try to choose another name or number"})
+        });
```

Comment:
This is a duplicate of lines 77 to 80, there could be a way to prevent this. Maybe define a function that would be placed inside the ```.catch``` for both cases? For example: 
```
function handle400Error(err) {
next({"status": 400, "messege": "Can't update contact, try to choose another name or number"})
}
```